op-e2e/actions/upgrades/isthmus_fork_test.go

```diff
@@ -394,3 +397,103 @@ func TestIsthmusNetworkUpgradeTransactions(gt *testing.T) {
+
+	log := testlog.Logger(t, log.LevelDebug)
+
+	// Activate Isthmus at genesis
+	dp.DeployConfig.ActivateForkAtGenesis(rollup.Isthmus)
+
+	require.NoError(t, dp.DeployConfig.Check(log), "must have valid config")
+
+	alloc := *helpers.DefaultAlloc
+	alloc.L2Alloc = make(map[common.Address]types.Account)
+
+	// Deploy EIP-7251 and EIP-7002 contracts
+	alloc.L2Alloc[params.WithdrawalQueueAddress] = types.Account{ // EIP-7002
+		Code:    params.WithdrawalQueueCode,
+		Nonce:   1,
+		Balance: new(big.Int),
+	}
+	alloc.L2Alloc[params.ConsolidationQueueAddress] = types.Account{ // EIP-7251
+		Code:    params.ConsolidationQueueCode,
+		Nonce:   1,
+		Balance: new(big.Int),
+	}
+
+	sd := e2eutils.Setup(t, dp, &alloc)
+
+	_, _, _, sequencer, engine, _, _, _ := helpers.SetupReorgTestActors(t, dp, sd, log)
+	ethCl := engine.EthClient()
+	signer := types.NewPragueSigner(new(big.Int).SetUint64(dp.DeployConfig.L2ChainID))
+
+	sequencer.ActL2StartBlock(t)
+
+	ret, err := ethCl.CallContract(context.Background(), ethereum.CallMsg{
+		To:   &params.WithdrawalQueueAddress,
+		Data: []byte{},
+	}, nil)
+
+	require.NoError(t, err)
+	fee := new(uint256.Int).SetBytes(ret)
+
+	// Send a transaction to the EIP-7251 contract
+	txdata := &types.DynamicFeeTx{
+		ChainID:   new(big.Int).SetUint64(dp.DeployConfig.L2ChainID),
+		Nonce:     0,
+		To:        &params.WithdrawalQueueAddress,
+		Gas:       500000,
+		Data:      make([]byte, 56),
+		Value:     fee.ToBig(),
+		GasFeeCap: new(big.Int).SetUint64(5000000000),
+		GasTipCap: new(big.Int).SetUint64(2),
+	}
+	tx := types.MustSignNewTx(dp.Secrets.Alice, signer, txdata)
+
+	err = ethCl.SendTransaction(t.Ctx(), tx)
+	require.NoError(gt, err, "failed to send withdrawal request tx")
+
+	_, err = engine.EngineApi.IncludeTx(tx, dp.Addresses.Alice)
+	require.NoError(gt, err, "failed to include tx")
+
+	ret, err = ethCl.CallContract(context.Background(), ethereum.CallMsg{
+		To:   &params.ConsolidationQueueAddress,
+		Data: []byte{},
+	}, nil)
+
+	require.NoError(t, err)
+	fee = new(uint256.Int).SetBytes(ret)
+
+	// Send a transaction to the EIP-7251 contract
+	txdata = &types.DynamicFeeTx{
+		ChainID:   new(big.Int).SetUint64(dp.DeployConfig.L2ChainID),
+		Nonce:     1,
+		To:        &params.ConsolidationQueueAddress,
+		Gas:       500000,
+		Data:      make([]byte, 96),
+		Value:     fee.ToBig(),
+		GasFeeCap: new(big.Int).SetUint64(5000000000),
+		GasTipCap: new(big.Int).SetUint64(2),
+	}
+	tx = types.MustSignNewTx(dp.Secrets.Alice, signer, txdata)
+
+	err = ethCl.SendTransaction(t.Ctx(), tx)
+	require.NoError(gt, err, "failed to send consolidation queue request tx")
+
+	_, err = engine.EngineApi.IncludeTx(tx, dp.Addresses.Alice)
+	require.NoError(gt, err, "failed to include tx")
+
+	sequencer.ActL2EndBlock(t)
+
+	// get receipt
+	receipt, err := ethCl.TransactionReceipt(context.Background(), tx.Hash())
+	require.NoError(t, err)
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status, "transaction must pass")
```

Comment:
Also good to assert that there aren't any logs emitted for both included transactions.

Reply:
I tried this, but I think this does produce a log if the contract is publicly deployed. It does seem to be hard to find info about this in the EIP, so I might be misunderstanding something.

Reply:
Yep, found this: https://github.com/ethereum/EIPs/pull/8890

Reply:
Yeah, what's way more critical to check is that the requests root stays empty.

Reply:
Speaking of - can we add a sanity check on the L2 block's header that the requests root is `sha256("")`?

Reply:
added!